backends/p4tools/common/lib/taint.cpp

```diff
@@ -30,22 +30,27 @@ static bitvec computeTaintedBits(const IR::Expression *expr) {
     if (expr->is<IR::SymbolicVariable>()) {
         return {};
     }
-
     if (const auto *taintExpr = expr->to<IR::TaintExpression>()) {
+        if (expr->type->is<IR::Type_Unknown>())
```

Comment:
I would add the failing two tests to Xfail instead of these changes. Looks like there are some expressions which are missing their type.

Reply:
The gtests create some "raw" expressions and don't run typechecking, so you end up with untyped constants that in turn result in values with Type_Unknown.  Another possible fix would be to fix the gtest code to create typed constants instead of untyped.

Reply:
I see, I went back to the failing gtest and looked at Taint09. Fixing the types there and using IR::Constant::get seems like a better fix.

Reply:
Adding the type properly in the gtest seems to be a better solution, so I did that.